gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/util/TemporalActivityUtils.java

```diff
@@ -0,0 +1,101 @@
+    activityConfigurationStrategies.put(ActivityType.RECOMMEND_SCALING, new ActivityConfigurationStrategy.RecommendScalingActivityConfigurationStrategy());
+    activityConfigurationStrategies.put(ActivityType.DELETE_WORK_DIRS, new ActivityConfigurationStrategy.DeleteWorkDirsActivityConfigurationStrategy());
+    activityConfigurationStrategies.put(ActivityType.PROCESS_WORKUNIT, new ActivityConfigurationStrategy.ProcessWorkunitActivityConfigurationStrategy());
+    activityConfigurationStrategies.put(ActivityType.COMMIT, new ActivityConfigurationStrategy.CommitActivityConfigurationStrategy());
+  }
+
+  /**
+   * Builds and returns an {@link ActivityOptions} object configured with the specified {@link ActivityType} and properties.
+   *
+   * @param activityType the type of the activity for which the options are being built.
+   * @param props the properties to be used for configuring the activity options.
+   * @return an {@link ActivityOptions} object configured with the specified activity type and properties.
+   */
+  public static ActivityOptions buildActivityOptions(ActivityType activityType, Properties props) {
+    return ActivityOptions.newBuilder()
+        .setStartToCloseTimeout(getStartToCloseTimeout(activityType, props))
+        .setRetryOptions(buildRetryOptions(activityType, props))
+        .build();
+  }
+
+  /**
+   * Retrieves the start to close timeout duration for a given {@link ActivityType} based on the provided properties.
+   *
+   * @param activityType the type of the activity for which the start to close timeout is being retrieved.
+   * @param props the properties to be used for configuring the timeout.
+   * @return the start to close timeout duration for the specified activity type.
+   */
+  private static Duration getStartToCloseTimeout(ActivityType activityType, Properties props) {
+    ActivityConfigurationStrategy activityConfigurationStrategy = activityConfigurationStrategies.get(activityType);
+    if (activityConfigurationStrategy == null) {
+      log.warn("No configuration strategy found for activity type {}. Using default start to close timeout.", activityType);
+      return ActivityConfigurationStrategy.defaultStartToCloseTimeout;
+    }
+    return activityConfigurationStrategy.getStartToCloseTimeout(props);
+  }
+
+  /**
+   * Builds and returns an {@link RetryOptions} object configured with the specified {@link ActivityType} and properties.
+   *
+   * @param activityType the type of the activity for which the options are being built.
+   * @param props the properties to be used for configuring the activity options.
+   * @return an {@link RetryOptions} object configured with the specified activity type and properties.
+   */
+  private static RetryOptions buildRetryOptions(ActivityType activityType, Properties props) {
+    // Currently returning just the default retry options for each activity type
+    return DEFAULT_RETRY_OPTIONS;
```

Comment:
retry settings are currently hardcoded which would requires a code change. Please use PropertiesUtils to fetch each parameter with default values